### PR TITLE
feat(admin): visualize the federation

### LINKED ISSUE #1788: 🚀 [Feature] Federation 1.4: Admin: Monitor federated communities and their status
## 🚀 Feature

We need a new Page in the admin showing the table contents of `community_federation` and their verification status and time.

This can be a simple list allowing us to monitor the process of federation.

(optional) have a autorefresh on that page to monitor the dynamic of the federation

### LINKED ISSUE #2595: 🚀 [Feature] [Admin] Visualize the federation
## 🚀 Feature

Duplicate Issue: #1788 

[Admin] Visualize the federation.

Make a new page in the menu called `Federation` visualizing the table `communities`.
It is important to format the date strings to a readable form ( `12 seconds ago`, ...).

Reuse the old Community resolver and remove/reuse stuff from it. The own Community name etc can be read from the config.

Refresh:
- Button for manuell Request
- Auto-Refresh optional

backend:  solved in issue #2800 
Query:
```
query {
    getCommunities {
      id
      foreign
      publicKey
      url
      lastAnnouncedAt
      verifiedAt
      lastErrorAt
      createdAt
      updatedAt
    }
  }
  ```

### LINKED ISSUE #2833: 🚀 [Feature] Admin: Style
## 🚀 Feature

As admin their is the impression that green color of the header and the blue color of open contributions seem not ordered and professional.
Expected would be a black or anthracite background. The open contribution in light gray.

## 🤖 Todo
- [x] Change background color from green to ????
- [ ] Change info color for contribution to ????

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
visualise the federation

- sorted by publicKeys
        oreign: 'ASC',
        createdAt: 'DESC',
        lastAnnouncedAt: 'DESC',

- Displayed 
       verified
      url
      publicKey (0, 26)
      lastAnnouncedAt
      createdAt

The table can be reloaded by hand
 
### Issues
- fixes #2595 
- fixes #2833 
- fixes #1788

image

https://user-images.githubusercontent.com/1324583/228506019-4f5b0eb4-c098-49ef-9c97-37e0cbd17b66.mp4

### CODE REVIEW

File: admin/src/components/Fedaration/FederationVisualizeItem.vue [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        default:
+          return en
+      }
+    },
+    lastAnnouncedAt() {
+      if (this.item.lastAnnouncedAt) {
+        return formatDistanceToNow(new Date(this.item.lastAnnouncedAt), {
+          includeSecond: true,
+          addSuffix: true,
+          locale: this.fnsLocale,
```
Comment (Mogge) [inline admin/src/components/Fedaration/FederationVisualizeItem.vue:59]:
```suggestion
          locale: locales[this.locale],
```

File: admin/src/components/Fedaration/FederationVisualizeItem.vue [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+export default {
+  name: 'FederationVisualizeItem',
+  props: {
+    item: { type: Object },
+  },
+  data() {
+    return {
+      oldPublicKey: '',
+      formatDistanceToNow,
```
Comment (Mogge) [inline admin/src/components/Fedaration/FederationVisualizeItem.vue:26]: What is this for?

**Review by Mogge — CHANGES_REQUESTED** [review]

File: admin/src/pages/FederationVisualize.vue [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import FederationVisualizeItem from '../components/Fedaration/FederationVisualizeItem.vue'
+
+export default {
+  name: 'FederationVisualize',
+  components: {
+    FederationVisualizeItem,
+  },
+  data() {
+    return {
+      oldPublicKey: '',
```
Comment (mahula) [inline admin/src/pages/FederationVisualize.vue:43]: What is `oldPublickey` for (it also occures once in FederationVisualizeItem.vue)?

File: admin/src/components/Fedaration/FederationVisualizeItem.vue [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    verified() {
+      return new Date(this.item.verifiedAt) >= new Date(this.item.lastAnnouncedAt)
+    },
+    icon() {
+      return this.verified ? 'check' : 'x-circle'
+    },
+    variant() {
+      return this.verified ? 'success' : 'danger'
+    },
+    fnsLocale() {
```
[the remark this answers is not in the document]
Comment (mahula) [reply admin/src/components/Fedaration/FederationVisualizeItem.vue]: Fixed with 206dcda7750ad1aebce63e44b746610c68224fed.

File: admin/src/components/Fedaration/FederationVisualizeItem.vue [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      </b-col>
+      <b-col cols="2">{{ lastAnnouncedAt }}</b-col>
+      <b-col cols="2">{{ createdAt }}</b-col>
+    </b-row>
+  </div>
+</template>
+<script>
+import { formatDistanceToNow } from 'date-fns'
+import { de, en, fr, es, nl } from 'date-fns/locale'
+
```
[the remark this answers is not in the document]
Comment (mahula) [reply admin/src/components/Fedaration/FederationVisualizeItem.vue:17]: Fixed with 206dcda7750ad1aebce63e44b746610c68224fed.

File: admin/src/components/Fedaration/FederationVisualizeItem.vue (hunk 1 above)
Comment (mahula) [reply admin/src/components/Fedaration/FederationVisualizeItem.vue]: Fixed with 206dcda7750ad1aebce63e44b746610c68224fed.

File: admin/src/components/Fedaration/FederationVisualizeItem.vue [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        })
+      }
+      return ''
+    },
+    createdAt() {
+      if (this.item.createdAt) {
+        return formatDistanceToNow(new Date(this.item.createdAt), {
+          includeSecond: true,
+          addSuffix: true,
+          locale: this.fnsLocale,
```
[the remark this answers is not in the document]
Comment (mahula) [reply admin/src/components/Fedaration/FederationVisualizeItem.vue]: Fixed with 206dcda7750ad1aebce63e44b746610c68224fed.

File: admin/src/pages/FederationVisualize.vue [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+      communities: [],
+      icon: '',
+      animation: '',
```
[the remark this answers is not in the document]
Comment (mahula) [reply admin/src/pages/FederationVisualize.vue]: Done in adcff96e293e87208d1e0fc017ab61c53af07ad9.

File: admin/src/pages/FederationVisualize.spec.js [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+        lastErrorAt: null,
+        createdAt: '2023-04-07T11:45:06.254Z',
+        updatedAt: null,
+        __typename: 'Community',
+      },
+      {
+        id: 1775,
+        foreign: true,
+        publicKey: 'c7ca9e742421bb167b8666cb78f90b40c665b8f35db8f001988d44dbb3ce8527',
+        url: 'http://localhost/api/api/1_1',
```
Comment (Elweyn) [inline admin/src/pages/FederationVisualize.spec.js:45]:
```suggestion
        url: 'http://localhost/api/1_1',
```

**Review by Elweyn — APPROVED** [review]

**Review by Elweyn — APPROVED** [review]

**Pull request merged**